mir_eval/separation.py

```diff
@@ -88,26 +88,18 @@ def validate(reference_sources, estimated_sources):
                          'source to be non-silent, having a silent estiamted '
                          'source will result in an underdetermined system.')
 
-    if estimated_sources.shape[0] > MAX_SOURCES:
+    if (estimated_sources.shape[0] > MAX_SOURCES or
+            reference_sources.shape[0] > MAX_SOURCES):
         raise ValueError('The supplied matrices should be of shape (n_sources,'
-                         ' n_samples) but estimated_sources.shape[0] = {} '
-                         'which is greater than '
+                         ' n_samples) but n_sources = {} which is greater than'
```

Comment:
This wording is less clear than stating that one of the arguments has the wrong shape, and we haven't defined "n_sources" anywhere (only nsrc).  Since we already checked that `estimated_sources.shape == reference_sources.shape`, if you want to collapse these two checks, instead of writing "n_sources", write "estimated_sources.shape[0] = {} and reference_sources.shape[0] = {}".